modules/twinkleblock.js

```diff
@@ -516,6 +516,17 @@ Twinkle.block.callback.change_action = function twinkleblockCallbackChangeAction
 		}
 	}
 
+	// grab discretionary sanctions list from en-wiki
+	var parent = this;
+	new Morebits.wiki.getCachedWikitext('Template:Ds/topics.json', 'Get JSON list of discretionary sanctions', parent, function(wikitext) {
+		var dsinfo = JSON.parse(wikitext);
+		var $select = $('[name="dstopic"]');
+		$.each(dsinfo, function(key, value) {
+			var $option = $('<option>').val(value.code).text(key).prop('label', key);
+			$select.append($option);
+		});
```

Comment:
A more efficient way to update the DOM is `$options = $.map(dsinfo, (k, v) => $('<option>').val(...)...)` then `$select.append($options)` (so that DOM is manipulated just once instead of `$options.length` number of times)